devops/jobs/CloudFlarePurgeCacheOrigin.groovy

```diff
@@ -0,0 +1,95 @@
+/*
+
+    Variables consumed from the EXTRA_VARS input to your seed job in addition
+    to those listed in the seed job.
+
+    Variables consumed for this job:
+        * CLOUDFLARE_EMAIL: (Required - email address)
+        * CLOUDFLARE_API_KEY:
+        * ORIGIN: (Required - jenkins name of git credentials)
+        * CLOUDFLARE_ZONE_ID: (Required - The CloudFlare Zone ID)
+        * TARGET_PATH: (Required - Authentication key for the account that would make API calls)
+        * CLOUDFLARE_SITE_URL: (Required - Email of the account for making API calls)
+
+*/
+package devops.jobs
+
+import static org.edx.jenkins.dsl.Constants.common_wrappers
+import static org.edx.jenkins.dsl.Constants.common_logrotator
+
+class CloudFlarePurgeCacheOrigin {
+    public static def job = { dslFactory, extraVars ->
+        dslFactory.job(extraVars.get("FOLDER_NAME","Monitoring") + "/cloudflare-purge-cache-origin") {
+            wrappers common_wrappers
+            logRotator common_logrotator
+
+            assert extraVars.containsKey('CLOUDFLARE_SITE_URL') : "Please provide Cloudflare site URL"
+            assert extraVars.containsKey('ORIGIN') : "Please provide Origin"
+
+            parameters {
+                stringParam("CLOUDFLARE_SITE_URL", extraVars.get('CLOUDFLARE_SITE_URL'),
+                        "Cloudflare site URL")
+                stringParam("ORIGIN", extraVars.get('ORIGIN'),"origin")
+
+
+                choiceParam("Site",
+                            ["edx-video.net",
+                             "edx-cdn.org",
+                             "edx.org",
+                            ],
+                            "Site for which you want to purge cache")
+            }
+
+            wrappers {
+                    credentialsBinding {
```

Comment:
I know it's less important in Groovy than Python, but if you wanted to you could match the new lines + indentation for code blocks and strings in the parameters and wrappers section.